### PR TITLE
Implement the POST request. Add the comment box to the webpage.

### PR DESCRIPTION
**Pull request by jiaxi312**
- Add the comment box to the webpage so that people can leave their comments.

- There is an option for submitting the comment anonymously without showing the name.

- If people type their name and choose the anonymous option, they will still submit their comments anonymously.

- The screen shoot for the current page is https://screenshot.googleplex.com/rv0SJueWPbE.png

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -35,6 +35,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the information from the post
+        String comments = getParameter(request, "comment-input", "");
+        String name = getParameter(request, "name-input", "Anonymous");
+        boolean isAnonymous = Boolean.parseBoolean(
+                                getParameter(request, "anonymous", "false"));
+        
+        // Check if the user want to submit the comment anonymously
+        if (isAnonymous) {
+            name = "Anonymous";
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You should use constant variables to store string like this.
```
static final String ANONYMOUS = "Anonymous";
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -35,6 +35,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         response.getWriter().println(json);
     }
 
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the information from the post
+        String comments = getParameter(request, "comment-input", "");
+        String name = getParameter(request, "name-input", "Anonymous");
+        boolean isAnonymous = Boolean.parseBoolean(
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: move this variable declaration to make it close to its usage.
```
bool isAnonymous = ...;
if (isAnonymous) {
}
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -35,6 +35,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+        response.setContentType("text/html");
+        response.getWriter().println(name + " leaves comment: ");
+        response.getWriter().println(comments);
+    }
+
+    /**
+     * @return the request parameter, or the default value if the parameter
+     *         was not specified by the client
+     */
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: consider to rename it to ```getParameterWithDefault```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (zixiyan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Actually I means
```
String name = ...

// Check if the user want to submit the comment anonymously
bool isAnonymous = ...;
if (isAnonymous) {
}
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -35,6 +35,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         response.getWriter().println(json);
     }
 
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        final String ANONYMOUS = "Anonymous";
```
Comment (floatfeather) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You may want to add a 'static' before this variable or change the naming. https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -50,7 +82,6 @@ private static String getInformationInJson() {
         information.add(new Date().toString());
 
         // convert it to json format and returns that json
```
Comment (floatfeather) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: 'Convert'?

**Review by floatfeather — CHANGES_REQUESTED** [review]

**Review by floatfeather — APPROVED** [review]

**Pull request merged**